frontend/spec/features/add_as_you_go_spec.rb

```diff
@@ -0,0 +1,379 @@
+# frozen_string_literal: true
+require 'spec_helper'
+require 'rails_helper'
+
+describe 'Add As You Go buttons', js: true do
+  before(:each) do
+    login_admin
+  end
+
+  context 'for non-Agent record types' do
+    shared_examples 'top-level non-Notes subforms' do |type, num_expected_subforms, num_expected_header_buttons|
+      title = type == 'location_batch' ? type : "#{type}s"
+      route = type == 'location_batch' ? '/locations/batch' : "/#{type}s/new"
+
+      it "for #{title}" do
```

Comment:
I think we are missing a context (or maybe a describe?) block here that could be somethig like: `context "when creating #{title}"`.

The it block, should give a hint of what is actually being asserted here. Maybe something like: `it "contains add-as-you-go buttons"